### ISSUE TITLE
strange log order

### ISSUE THREAD

**Issue by electronick** [miguelgrinberg/python-socketio#1173]
I'm  using aiohttp + python-socketio, and I see strange behavior when stopping the server. Basically, the server doesn't shutdown until the last wait_task (engineio/asyncio_socket.py:213) finished. Also I see strange log sequence of some aiohttp.access logs related to upgrading socket.io connection, however I don't use aiohttp.access logs for socket.io part, I use only aiohttp.server and aiohttp.ws.

Configuration is the following:
MacOSX

where client side:
socket.io-client 4.6.1

and server side:
Python 3.10
aiohttp 3.8.4
python-socketio 5.8.0

Created demo app to demonstrate:
```python
import asyncio
import logging

from aiohttp import web
from aiohttp.log import server_logger, ws_logger
from socketio import AsyncServer


async def home_handler(request):
    return web.Response(text="Hello, world")


async def make_app():
    app = web.Application()
    logging.basicConfig(level=logging.DEBUG)

    app.router.add_route('GET', '/home', home_handler)

    sio = AsyncServer(async_mode='aiohttp', engineio_logger=ws_logger, logger=server_logger)

    @sio.event
    async def connect(sid, environ):
        ws_logger.info('connect %s', sid)

    @sio.event
    async def disconnect(sid):
        ws_logger.info('disconnect %s', sid)

    sio.attach(app)
    app['sio'] = sio
    return app


async def on_shutdown(app):
#    await app['sio'].eio.disconnect()
    raise asyncio.CancelledError


async def main() -> web.Application:
    app = await make_app()
    app.on_shutdown.append(on_shutdown)
    return app


if __name__ == "__main__":
    web.run_app(main(), host='localhost', port=8888)
```

To reproduce just comment line 36 with `raise asyncio.CancelledError`, without this line server hangs and stopped only after last wait_task (engineio/asyncio_socket.py:213) is finished or after killing server.

With uncommented line when I start app, then wait a bit while websocket connection upgrades and do KeyboardInterrupt I got the following weird log sequence:

```text
nick@book server % venv/bin/python start_aio_demo.py
INFO:aiohttp.websocket:Server initialized for aiohttp.
======== Running on http://localhost:8888 ========
(Press CTRL+C to quit)
INFO:aiohttp.websocket:12PbpRrUGotY351EAAAA: Sending packet OPEN data {'sid': '12PbpRrUGotY351EAAAA', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 25000}
INFO:aiohttp.access:127.0.0.1 [07/Apr/2023:01:52:04 +0000] "GET /socket.io/?EIO=4&transport=polling&t=OTPQUJM HTTP/1.1" 200 289 "http://loc.loc:8080/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
INFO:aiohttp.websocket:12PbpRrUGotY351EAAAA: Received packet MESSAGE data 0
INFO:aiohttp.websocket:connect 9FjTbnCyX9UAlUurAAAB
INFO:aiohttp.websocket:12PbpRrUGotY351EAAAA: Sending packet MESSAGE data 0{"sid":"9FjTbnCyX9UAlUurAAAB"}
INFO:aiohttp.access:127.0.0.1 [07/Apr/2023:01:52:04 +0000] "POST /socket.io/?EIO=4&transport=polling&t=OTPQUJb&sid=12PbpRrUGotY351EAAAA HTTP/1.1" 200 233 "http://loc.loc:8080/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
INFO:aiohttp.websocket:12PbpRrUGotY351EAAAA: Received request to upgrade to websocket
INFO:aiohttp.access:127.0.0.1 [07/Apr/2023:01:52:04 +0000] "GET /socket.io/?EIO=4&transport=polling&t=OTPQUJc&sid=12PbpRrUGotY351EAAAA HTTP/1.1" 200 224 "http://loc.loc:8080/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
INFO:aiohttp.access:127.0.0.1 [07/Apr/2023:01:52:04 +0000] "GET /socket.io/?EIO=4&transport=polling&t=OTPQUJk&sid=12PbpRrUGotY351EAAAA HTTP/1.1" 200 192 "http://loc.loc:8080/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
INFO:aiohttp.websocket:12PbpRrUGotY351EAAAA: Upgrade to websocket successful
INFO:aiohttp.websocket:12PbpRrUGotY351EAAAA: Sending packet PING data None
INFO:aiohttp.websocket:12PbpRrUGotY351EAAAA: Received packet PONG data 
INFO:aiohttp.access:127.0.0.1 [07/Apr/2023:01:52:35 +0000] "GET /home HTTP/1.0" 200 164 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
^CINFO:aiohttp.websocket:service task canceled
INFO:aiohttp.websocket:disconnect 9FjTbnCyX9UAlUurAAAB
INFO:aiohttp.access:127.0.0.1 [07/Apr/2023:01:52:04 +0000] "GET /socket.io/?EIO=4&transport=websocket&sid=12PbpRrUGotY351EAAAA HTTP/1.1" 101 0 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
```

Last line in the log should be posted much earlier  (with 101 upgrade status code) and is shown only when server stops. As you could see it has timestamp of 07/Apr/2023:01:52:04 comparing to a timestamp of my visit of  /home at 07/Apr/2023:01:52:35, but appended to log much later and only after server exit. Not sure if this log sequence is connected to freeze issue, but who knows.

Could you take a look please?

**Comment by miguelgrinberg** [comment]
A clean shutdown is in my list of tasks for the short term. There is currently some tasks that aren't properly shutdown when you stop the server, so hangs and crashes are expected.

I have no control over the logging options in aiohttp, so I cannot really address aiohttp logs that are delayed or out of order.

**Comment by miguelgrinberg** [comment]
Shutdown has been implemented in release 5.10.
